File: exercise1.cpp
#include <iostream>

double calculatePurity(int sensorValue) {
    if (sensorValue < 0 || sensorValue > 24) {
        std::cout << "Invalid sensor value. It should be between 0 and 24 Karats." << std::endl;
        return -1;
    }
    return (sensorValue / 24.0) * 100;
}

int main() {
    int sensorValue;
    std::cout << "Enter the sensor value in Karats: ";
    std::cin >> sensorValue;

    double purity = calculatePurity(sensorValue);
    if (purity != -1) {
        std::cout << "The purity of the gold is: " << purity << "%" << std::endl;
    }

    return 0;

File: exercises/python/exercise1.py
def calculate_purity(sensor_value):
    if sensor_value < 0 or sensor_value > 24:
        print("Invalid sensor value. It should be between 0 and 24 Karats.")
        return -1
    return (sensor_value / 24.0) * 100


def main():
    sensor_value = int(input("Enter the sensor value in Karats: "))
    purity = calculate_purity(sensor_value)
    if purity != -1:
        print(f"The purity of the gold is: {purity}%")


if __name__ == "__main__":
    main()
